File: codes/dart/object_entries.dart
import "dart:convert";

void main() {
    dynamic jsonStringify(dynamic anything, { bool pretty = false, String indent = "    " }) {
		if (pretty == true) {
			final jsonEncoder = new JsonEncoder.withIndent(indent);
			return jsonEncoder.convert(anything);
		}
		final jsonEncoder = new JsonEncoder();
		return jsonEncoder.convert(anything).split(',').join(', ');
    }

	// There's no JavaScript-like Object.entries() in Dart.
    // But, we can create our own function to mimic it in Dart.

	dynamic objectEntries(dynamic anObject) {
        // JavaScript-like Object.entries() function
        dynamic newArray = <dynamic>[];
		anObject.entries.toList().asMap().forEach((dynamic objectEntryIndex, dynamic objectEntry) {
			final objectKey = objectEntry.key;
			final objectValue = objectEntry.value;
			newArray.add(<dynamic>[objectKey, objectValue]);
		});
        return newArray;
    }

	print("\n// JavaScript-like Object.entries() in Dart Map");

    final friend = <String, dynamic>{
        "name": "Alisa",
        "country": "Finland",
        "age": 25,
    };
    print("friend: ${jsonStringify(friend, pretty: true)}");

	print("// using JavaScript-like Object.entries() function \"objectEntries\"");

	print("friend entries: ${jsonStringify(objectEntries(friend))}");
	// friend entries: [["name", "Alisa"], ["country", "Finland"], ["age", 25]]
}

File: codes/php/object_entries.php
<?php

// There's no JavaScript-like Object.entries() in PHP.
// But, we can create our own function to mimic it in PHP.

function object_entries($an_object) {
    // JavaScript-like Object.entries() function
    $new_array = [];
    foreach ($an_object as $object_key => $object_value) {
        $new_array = [...$new_array, [$object_key, $object_value]];
    }
    return $new_array;
};

echo("\n// JavaScript-like Object.entries() in PHP associative-array" . "\n");

$friend = [
    "name" => "Alisa",
    "country" => "Finland",
    "age" => 25
];
echo("friend: " . str_replace("/\n$/", "", json_encode($friend, JSON_PRETTY_PRINT)) . "\n");

echo("// using JavaScript-like Object.entries() function \"object_entries\"" . "\n");

echo("friend entries: " . str_replace(",", ", ", str_replace(":", ": ", json_encode(object_entries($friend)))) . "\n");
// friend entries: [["name", "Alisa"], ["country", "Finland"], ["age", 25]]
